packages/@adobe/spectrum-css-temp/components/steplist/index.css

```diff
@@ -131,137 +74,177 @@ governing permissions and limitations under the License.
-  white-space: normal;
-
-  transform: translateX(-50%);
+.spectrum-Steplist-item > a {
+  grid-area: link;
+  display: grid;
+  justify-items: center;
+  align-items: center;
+  grid-template-areas:
+    "number label";
+  
+  .spectrum-Steplist-marker {
+    grid-area: number;
+    outline: none;
+    box-sizing: border-box; /* So we can stroke nicely */
+    border-style: solid;
+    border-width: var(--spectrum-steplist-marker-border-width);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+  .spectrum-Steplist-label {
+    grid-area: label;
+  }
 }
 
-.spectrum-Steplist-markerContainer {
-  display: block;
-
-  z-index: 2;
-  position: absolute;
-  bottom: -10px;
-  left: 50%;
-
-  width: var(--spectrum-steplist-marker-hitArea);
-  height: var(--spectrum-steplist-marker-hitArea);
-
-  margin-left: calc(calc(var(--spectrum-steplist-marker-hitArea) / 2) * -1);
+.spectrum-Steplist--vertical {
+  .spectrum-Steplist-chevron {
+    display: none;
+  }
 }
 
-/* Circle step marker */
-.spectrum-Steplist-marker {
-  box-sizing: border-box; /* So we can stroke nicely */
-  display: block;
-
-  position: absolute;
-  top: 50%;
-  left: 50%;
-  margin-top: calc(calc(var(--spectrum-steplist-marker-diameter) / 2) * -1);
-  margin-left: calc(calc(var(--spectrum-steplist-marker-diameter) / 2) * -1);
-
-  width: var(--spectrum-steplist-marker-diameter);
-  height: var(--spectrum-steplist-marker-diameter);
-
-  border-radius: var(--spectrum-steplist-marker-diameter);
+/* Generated item counts on all sizes except small */
+.spectrum-Steplist--medium,
+.spectrum-Steplist--large,
+.spectrum-Steplist--xlarge {
+  counter-reset: listItemCount;
+  li {
+    counter-increment: listItemCount;
+  }
+  .spectrum-Steplist-item {
+    &.is-complete .spectrum-Steplist-marker::before,
+    &.is-selected .spectrum-Steplist-marker::before {
+      content: counter(listItemCount);
+    }
 
-  border: 2px solid;
-  border-color: transparent;
-  background: transparent;
+    &.is-hovered {
+      &.is-navigable .spectrum-Steplist-marker::before {
+        content: counter(listItemCount);
+      }
+    }
+  }
 }
 
-/* Step line */
-.spectrum-Steplist-segment {
-  display: block;
-  z-index: 1;
-  position: absolute;
-  right: calc(var(--spectrum-steplist-step-width) - calc(var(--spectrum-steplist-marker-diameter) * 1.5));
-
-  box-sizing: content-box;
-  width: calc(calc(var(--spectrum-steplist-step-width) * 1.5) - calc(var(--spectrum-steplist-marker-diameter) * 2));
-  bottom: calc(calc(var(--spectrum-steplist-segment-height) / 2) * -1);
-
-  /* Default is dashed */
-  border-bottom-width: var(--spectrum-steplist-segment-height);
-  border-bottom-style: dashed;
+/* small size */
+.spectrum-Steplist--small {
+  .spectrum-Steplist-item {
+    grid-template-columns: calc(var(--spectrum-steplist-marker-diameter-small) * 3) auto;
+    margin-left: calc(var(--spectrum-steplist-marker-diameter-small));
```

Comment:
using grid, should this actually be
```
gap: calc(var(--spectrum-steplist-marker-diameter-small));
```

if it really needs to be margin, then we need to use the logical property `margin-inline-start` instead of `margin-left`

Reply:
Yeah, I used margin because it's the space between one steplist item and the next, not the gap between cells within the steplist item.  I'll switch them to `margin-inline-start`